protocols/kad/dht.proto

```diff
@@ -18,6 +18,14 @@ message Record {
 
 	// Time the record was received, set by receiver
 	string timeReceived = 5;
+
+    // The original publisher of the record.
+    // Currently specific to rust-libp2p.
```

Comment:
Well, I'm not a fan of this. :upside_down_face:

Reply:
That's ok, neither am I. Unfortunately I only know of one alternative that avoids protocol additions but is a step backwards w.r.t. Kademlia proper:

  1. Removing (re-)replication, leaving only re-publication of records.
  2. The `publisher` field of a record would (for now) only ever be either `None` or `Some(local_id)` if the local node considers itself the publisher. That is, every node only knows about which records it is itself the publisher of, but not who the alleged publisher of other records is.

Albeit a step backwards (from my perspective at least), it wouldn't preclude (re-)adding replication later, if the libp2p "specs" ever accommodate for it (I'm not holding my breath). Would that indeed be your preference or do you even see other options?

Reply:
Let's go with that.

Reply:
Do you have a final verdict on the matter? It may be worth mentioning that, despite these new fields, the protocol is still compatible with other libp2p Kademlia implementations, at least that is my intention. The protocol is also easy to (re-)adjust later, I think.

Reply:
I'm fine with doing that, for the exact reasons you mentioned.